mit-press-direct/parser.js

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env node
+
+'use strict';
+const Parser = require('../.lib/parser.js');
+
+/**
+ * Recognizes the accesses to the platform MIT Press Direct
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  let result = {};
+  let path   = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  // let param = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  let match;
+
+  if ((match = /^\/([a-z]+)\/article\/([0-9]+)\/([0-9]+)\/([0-9]+)\/[0-9]+\/([a-z0-9-]+)$/i.exec(path)) !== null) {
+    // https://direct.mit.edu/qss/article/1/2/675/96133/Citation-concept-analysis-CCA-of-Robert-K-Merton-s?searchresult=1
+    result.rtype    = 'ARTICLE';
+    result.mime     = 'HTML';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.spage = match[4];
+    result.unitid = match[1] + '-' + match[2] + '-' + match[3] + '-' + match[4];
+
+  } else if  ((match = /^\/([a-z]+)\/article-pdf\/([0-9]+)\/([0-9]+)\/([0-9]+)\/[0-9]+\/(.+)$/i.exec(path)) !== null) {
+    // https://direct.mit.edu/qss/article-pdf/1/2/675/1885761/qss_a_00029.pdf&hl=en&sa=T&oi=ucasa&ct=ufr&ei=RCVPY5vmCpCXywTDn5S4DA&scisig=AAGBfm24IIAYBlO-We0ogpow1Ai-IhTG2g
+    result.rtype = 'ARTICLE';
+    result.mime = 'PDF';
+    result.title_id = match[1];
+    result.vol = match[2];
+    result.issue = match[3];
+    result.spage = match[4];
+    result.unitid = match[1] + '-' + match[2] + '-' + match[3] + '-' + match[4] + '-pdf';
```

Comment:
It is not necessary to add '-pdf' for unitid, you should delete that in parser and in the analysis

Reply:
If I don't include the "-pdf" then the events for the HTML and PDF article will look the same and potentially deduplicate

Reply:
No, because the unitid is unique for each regular expression. For example, it will not be in conflict with the regular expression on line 34 because it's a other type of consultation.